include/nil/crypto3/block/detail/block_stream_processor.hpp

```diff
@@ -49,24 +48,44 @@ namespace nil {
             private:
                 constexpr static const std::size_t length_bits = params_type::length_bits;
                 // FIXME: do something more intelligent than capping at 64
-                constexpr static const std::size_t length_type_bits =
-                    length_bits < input_block_bits ? input_block_bits : length_bits > 64 ? 64 : length_bits;
-                typedef typename boost::uint_t<length_type_bits>::least length_type;
+                //constexpr static const std::size_t length_type_bits =
+                    //length_bits < input_block_bits ? input_block_bits : length_bits > 64 ? 64 : length_bits;
+                typedef typename boost::uint_t<64>::least length_type;
```

Comment:
Do we really need to fix the length to 64 bit? The length could represented with bits amount smaller than 64 bits.

I would really like for us to have this done as ```std::conditional``` between ```boost::uint_t<Size>``` and Boost.Multiprecision ```FixedPrecisionNumber``` concept. We can potentially have a VERY LONG message being encrypted.

Reply:
Changed now to previous version with 64 being replaced with sizeof(boost::uintmax_t) * CHAR_BIT.